### ISSUE TITLE
Fix S2589 FP: Pre or postfix increment or decrement operators

### ISSUE THREAD

**Issue by sit-md** [SonarSource/sonar-dotnet#9014]
### Description

S2589 is being triggered when using pre and postfix increment or decrement operators. When using compound operators, the same resulting operation does not trigger the warning (as is expected).

### Repro steps

False-Positive with in-/decrementor:
```
public void Test()
{
    int retry = 3;
    while (true)
    {
        retry--;
        if (retry == 0)
        {
            // Do something
        }
    }
}
```

### Expected behavior

No finding with compound assignment (expected):
```
public void Test()
{
    int retry = 3;
    while (true)
    {
        retry -= 1;
        if (retry == 0)
        {
            // Do something
        }
    }
}
```

### Actual behavior

"S2589 Change this condition so that it does not always evaluate to 'False'" is flagged on the if-statement.

### Known workarounds

Using compound operators instead.

### Related information

* SonarQube Enterprise Edition Version 10.3 (build 82913)
* C# Plugins version: 9.13.0.79967
* VS2022, latest
* MSBuild bundled with VS2022 / .NET 6
* SonarScanner for MSBuild 5.15
* Windows 2022

**Comment by cristian-ambrosini-sonarsource** [comment]
Hi @sit-md!
I'm not able to reproduce it with the snippets you provided running the latest version of the analyzer.
Please upgrade your plugin to version [9.23.1](https://github.com/SonarSource/sonar-dotnet/releases/tag/9.23.1.88495), this may resolve the issue you're experiencing.
If the issue persists, please provide a snippet where the issue is raised. 

Thanks!

**Comment by sit-md** [comment]
Hi @cristian-ambrosini-sonarsource,
Thank you for checking this so quickly! I forwarded the update inquiry to our SonarQube OPS team and will report again here once that is done and I could re-check the behavior.

**Comment by CristianAmbrosini** [comment]
Closing as not reproduced, please open a new ticket if you can reproduce it with the latest version of the analyzer.
